todo_app/data/db.py

```diff
@@ -0,0 +1,88 @@
+from os import getenv
+from typing import List
+from bson import ObjectId
+
+import pymongo
+
+from todo_app.data.mongo_item import MongoItem
+
+# Define a custom sorting key function
+def custom_sorting_key(doc):
+    # If due_date is None, return a value that comes last in sorting
+    if doc['due_date'] is None:
+        return float('inf')  # A very large number
+    else:
+        return doc['due_date']
+
+class DB:
+    def __init__(self):
+        self.client = pymongo.MongoClient(getenv('MONGO_CONNECTION_STRING'))
+        self.db = self.client[getenv('MONGO_DATABASE_NAME')]
+        self.collection = self.db['todos']
+
+    def get_items(self) -> List[MongoItem]:
+        arr = []
+        pipeline = [
+            {
+                "$addFields": {
```

Comment:
I'm not convinced this is really working, nor entirely sure what the purpose behind it is! Doesn't seem to cause any issues, but I can't see any evidence that empty due dates are being set to the year 3000..

Reply:
> I'm not convinced this is really working, nor entirely sure what the purpose behind it is! Doesn't seem to cause any issues, but I can't see any evidence that empty due dates are being set to the year 3000..

I wanted to order by due date ASC and have the items without due dates last. I couldn't find a simpler way to do this when some of the values were null. It adds a new field called '_due_date', sets it to the due date if one is set, otherwise it sets it to the year 3000. 

The screenshot below shows an item in the debugger that has a None due_date with the _due_date set to the year 3000:

![image](https://github.com/CameronB9/DevOps-Course-Starter/assets/135842865/cde24e2c-6c95-4d60-bab3-21ef1adbf10c)

It doesn't make it into the MongoItem class as I use it just for ordering. If you change the $sort part of the pipeline from 1 to -1 the ordering is reversed. I think it is working as I intended?

I can't remember exactly what happened when I tried to sort by due_date without the pipeline, I just remember that it didn't work.